packages/flutter/lib/src/widgets/two_dimensional_viewport.dart

```diff
@@ -894,9 +892,112 @@ abstract class RenderTwoDimensionalViewport extends RenderBox implements RenderA
   }
 
   @override
-  RevealedOffset getOffsetToReveal(RenderObject target, double alignment, { Rect? rect }) {
-    // TODO(Piinks): Add this back in follow up change (ensureVisible), https://github.com/flutter/flutter/issues/126299
-    return const RevealedOffset(offset: 0.0, rect: Rect.zero);
+  RevealedOffset getOffsetToReveal(
+    RenderObject target,
+    double alignment, {
+    Rect? rect,
+    AxisDirection? axisDirection,
+  }) {
+    // We must know which axis we are revealing for.
+    assert(axisDirection != null);
```

Comment:
It's odd that the parameter is typed as nullable, but then the first check is whether this isn't null... It should at least be documented, but ideally we'd avoid misleading API like that...

Reply:
It is nullable for 1D cases. The goal here was to make the static Scrollable.ensureVisible work for both 1D and 2D without needing to have multiple methods for different types of scrollables.